ValveResourceFormat/Resource/ResourceTypes/Morph.cs

```diff
@@ -0,0 +1,215 @@
+                    {
+                        var bundleType = int.Parse(bundleData.Key);
+                        var bundle = bundleData.Value as IKeyValueCollection;
+                        var rectU = (int)Math.Round(bundle.GetFloatProperty("m_flULeftSrc") * texWidth, 0);
+                        var rectV = (int)Math.Round(bundle.GetFloatProperty("m_flVTopSrc") * texHeight, 0);
+                        var ranges = bundle.GetFloatArray("m_ranges");
+                        var offsets = bundle.GetFloatArray("m_offsets");
+
+                        for (var row = rectV; row < rectV + rectHeight; row++)
+                        {
+                            for (var col = rectU; col < rectU + rectWidth; col++)
+                            {
+                                var colorIndex = row * texWidth + col;
+                                var color = texPixels[colorIndex];
+                                var dstI = row - rectV + yTopDst;
+                                var dstJ = col - rectU + xLeftDst;
+                                rectData[bundleType, dstI, dstJ] = new Vector3(
+                                    color.Red / 255f * ranges[0] + offsets[0],
+                                    color.Green / 255f * ranges[1] + offsets[1],
+                                    color.Blue / 255f * ranges[2] + offsets[2]
+                                    );
+                            }
+                        }
+                    }
+                }
+
+                var flexDataWithType = new Dictionary<int, List<byte>>();
+                for (int c = 0; c < bundleTypeCount; c++)
+                {
+                    var tmp = new List<byte>(width * height * BytesPerVertex);
+                    for (int i = 0; i < height; i++)
+                    {
+                        for (int j = 0; j < width; j++)
+                        {
+                            tmp.AddRange(BitConverter.GetBytes(rectData[c,i,j].X));
+                            tmp.AddRange(BitConverter.GetBytes(rectData[c,i,j].Y));
+                            tmp.AddRange(BitConverter.GetBytes(rectData[c,i,j].Z));
+                        }
+                    }
+                    flexDataWithType.Add(c, tmp);
+                }
+                _flexData.Add(morphName, flexDataWithType);
+            }
+        }
+
+        //TODO:Other enums are still unknown.
+        public bool CheckBundleId(int bundleId)
+        {
+            var bundleType = _bundleTypes.GetProperty<object>(bundleId.ToString());
+            if (bundleType is uint bundleTypeEnum)
+            {
+                if (bundleTypeEnum == (uint)MorphBundleType.MORPH_BUNDLE_TYPE_POSITION_SPEED)
+                {
+                    return true;
+                }
+            }
+            else if (bundleType is string bundleTypeString)
+            {
+                if (bundleTypeString == MorphBundleType.MORPH_BUNDLE_TYPE_POSITION_SPEED.ToString())
```

Comment:
Better to compare by string here.

And enum values to match style of other enums, like `PositionSpeed`.